oarepo_vocabularies/ui/theme/assets/semantic-ui/js/oarepo_vocabularies_ui/form/components/VocabularySelectField/VocabularySelectField.jsx

```diff
@@ -1,48 +1,64 @@
 import React from "react";
 import { Breadcrumb } from "semantic-ui-react";
-import { I18nString, RelatedSelectField } from "@js/oarepo_ui";
+import {
+  I18nString,
+  RelatedSelectField,
+  getTitleFromMultilingualObject,
+} from "@js/oarepo_ui";
 import _join from "lodash/join";
 import PropTypes from "prop-types";
+import { search } from "@js/oarepo_vocabularies";
 
 export const serializeVocabularySuggestions = (suggestions) =>
   suggestions.map((item) => {
-    const hierarchy = item.hierarchy.ancestors_or_self;
-    const key = _join(hierarchy, ".");
-    const sections = [
-      ...hierarchy.map((id, index, { length }) => ({
-        key: id,
-        content:
-          index === 0 ? (
-            <I18nString value={item.hierarchy.title[index]} />
-          ) : (
-            <span style={{ opacity: "0.5", fontSize: "0.8rem" }}>
+    const hierarchy = item?.hierarchy?.ancestors_or_self;
+    let sections;
+    let key = item.id;
+    if (hierarchy?.length > 1) {
+      key = _join(hierarchy, ".");
+      sections = [
+        ...hierarchy.map((id, index, { length }) => ({
+          key: id,
+          content:
+            index === 0 ? (
               <I18nString value={item.hierarchy.title[index]} />
-            </span>
+            ) : (
+              <span className="ui breadcrumb vocabulary-parent-item">
+                <I18nString value={item.hierarchy.title[index]} />
+              </span>
+            ),
+        })),
+      ];
+    }
+    if (typeof item === "string") {
+      return {
+        text: item,
+        value: item,
+        key: item,
+        name: item,
+        id: item,
+      };
+    } else {
+      return {
+        text:
+          hierarchy.length > 1 ? (
+            <Breadcrumb key={key} icon="left angle" sections={sections} />
+          ) : (
+            <I18nString value={item.title} />
           ),
-        active: index === 0 && length !== 1,
-      })),
-    ];
-    return {
-      text: <Breadcrumb key={key} icon="left angle" sections={sections} />,
-      value: item,
-      key: key,
-    };
+        value: item.id,
+        key: key,
+        data: item,
+        id: item.id,
+        title: item.title,
+        name: getTitleFromMultilingualObject(item?.title),
+      };
+    }
   });
 
-export const serializeVocabularyItem = (item, includeProps = ["id"]) => {
-  if (typeof item === "string") {
-    return { id: item };
-  } else if (Array.isArray(item)) {
-    return item.map((i) => serializeVocabularyItem(i));
-  } else {
-    return item;
-  }
-};
-
-export const deserializeVocabularyItem = (item) => {
-  return Array.isArray(item)
-    ? item.map((item) => deserializeVocabularyItem(item))
-    : item;
+// for adding free text items
+const serializeAddedValue = (value) => {
```

Comment:
Duplicit of `serializeVocabularySuggestions(value)` where value is just a string?

Reply:
@mirekys This is a specific function, that is applied to items that are added by the user, so not existing in what was for example provided by the API or similar. When you use allowAdditions on the dropdown, you can write any textual value (what we need for the organzational part of creatibutor), but as dropdown only works with an option that has that SUI shape i.e. key, value, text, if you wish to add this free text, then you also need to add this as an option and this is internally used by remoteselectfield, to serialize these, user added items.

Reply:
Understood, what I meant was, if this function doesn't do the same thing as what serializeVocabularySuggestions("my textual value") will do

Reply:
@mirekys Ah sorry, I misunderstood. The problem is that when you are serializing suggestions, suggestions are an array and you map over that array (serializeVocabularySuggestions), but in case of adding you are always adding one item at the time, in other words, if I pass serializeVocabularySuggestions to serializeAddedValue prop of RelatedSelectField, the addition stops working.